avm/res/web/site/main.bicep

```diff
@@ -234,6 +234,10 @@ var formattedRoleAssignments = [
   })
 ]
 
+// Validation: serverFarmResourceId is required for all app kinds except containerized function apps
+var isContainerizedFunctionApp = contains(kind, 'azurecontainerapps')
+var _ = isContainerizedFunctionApp ? null : (!empty(serverFarmResourceId) ? null : error('serverFarmResourceId is required for non-containerized function apps. For containerized function apps with Azure Container Apps, serverFarmResourceId can be omitted.'))
```

Comment:
Should not this be fail instead of error ? https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-flow-control